### PR TITLE
(#1024) Tee input fix

### LINKED ISSUE #1024: TeeInput might leak
Consider the following Code from the examples section of the README:

```java
new LengthOf(
  new TeeInput(
    "Hello, world!",
    new File("/code/a.txt")
  )
).intValue();
```

Because of https://github.com/yegor256/cactoos/pull/820 the `OutputStream` obtained from `File` will not be closed. It's also impossible for us to close the stream manually because the reference is lost. We just leave the resource open for the rest of the application.

I hit the problem in https://github.com/yegor256/takes/pull/942
```java
try (TempFile file = new TempFile()) {
  final Bytes body = new Utf8String("URL returnsCorrectLength!");
  new LengthOf(new TeeInput(body, file.value())).intValue();
  MatcherAssert.assertThat(
    "Body content of Body.Url doesn't have the correct length",
    new LengthOf(
      new Body.Url(file.value().toUri().toURL())
      ).intValue(),
      new IsEqual<>(body.asBytes().length)
    );
  }
}
```

The code runs fine on linux but fails on windows with:
```
java.nio.file.FileSystemException: 
C:\Users\appveyor\AppData\Local\Temp\1\4539165525569174340: The process cannot access the file because it is being used by another process.
```

We should at least fix the leaking example in the README and also add some warnings to the documentation of TeeInput.

### PR DESCRIPTION
**Pull request by aivinog1**
This PR fixes #1024. Reverting changes in #820 and add more javadocs with **WARNING**.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1149](https://github.com/yegor256/cactoos/issues/1149) is now in scope, role is `REV`

**Comment by aivinog1** [comment]
@0crat wait #1148

**Comment by 0crat** [comment]
> @0crat wait #1148 ([here](https://github.com/yegor256/cactoos/issues/1149#issuecomment-505404654))

@aivinog1 Job `gh:yegor256/cactoos#1149` is not assigned, can't get performer

**Comment by 0crat** [comment]
This pull request [#1149](https://github.com/yegor256/cactoos/issues/1149) is assigned to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), here is [why](https://www.0crat.com/footprint/C63314D6Z/74b48c86-c6e9-4e82-8795-90e27db96b80); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by aivinog1** [comment]
@0crat continue

**Comment by 0crat** [comment]
> @0crat continue ([here](https://github.com/yegor256/cactoos/issues/1149#issuecomment-505561635))

@aivinog1 Job [#1149](https://github.com/yegor256/cactoos/issues/1149) is not on hold, no impediment to remove

**Comment by aivinog1** [comment]
@llorllale Done.

File: src/test/java/org/cactoos/io/TeeInputStreamTest.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public void leftInputUnclosed() {
             MatcherAssert.assertThat(
                 "Can't use output after usage from TeeInput",
                 write.isClosed(),
-                new IsEqual<>(false)
+                new IsEqual<>(true)
```
Comment (llorllale) [inline src/test/java/org/cactoos/io/TeeInputStreamTest.java]: @aivinog1 please rename this test accordingly

Comment (aivinog1) [reply src/test/java/org/cactoos/io/TeeInputStreamTest.java]: Done

**Comment by aivinog1** [comment]
@llorllale I've fixed all your review comments.

**Comment by llorllale** [comment]
@rultor merge

**Comment by llorllale** [comment]
@aivinog1 check your commit msgs:

```
da0e90c: subject does not match regex [^\(#[0-9]+\) [^\s].{13,50}]
015810b: subject does not match regex [^\(#[0-9]+\) [^\s].{13,50}]
7ca784c: subject does not match regex [^\(#[0-9]+\) [^\s].{13,50}]
```

**Comment by aivinog1** [comment]
@llorllale Thanks. I've fixed my commits, let's try a merge again.

**Comment by llorllale** [comment]
@aivinog1 please squash your commits. Messages like "revert unnecessary changes" just add noise to the commit history

**Comment by aivinog1** [comment]
@llorllale Okay, I'll squash commits.

**Comment by aivinog1** [comment]
@llorllale I've squashed my commits. Can you check this request again, please?

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Code review was too long (15 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Job [#1149](https://github.com/yegor256/cactoos/issues/1149) is not in the agenda of @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), can't inspect

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1149](https://github.com/yegor256/cactoos/issues/1149) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Pull request merged**